packages/data-stores/src/starter-designs-queries/use-starter-designs-query.ts

```diff
@@ -0,0 +1,93 @@
+import { stringify } from 'qs';
+import { useQuery, UseQueryResult, QueryOptions } from 'react-query';
+import wpcomRequest from 'wpcom-proxy-request';
+import type {
+	StarterDesignsGenerated,
+	StarterDesignsGeneratedQueryParams,
+	AllStarterDesigns,
+} from './types';
+import type { Category, Design, DesignRecipe } from '@automattic/design-picker/src/types';
+
+interface Options extends QueryOptions< AllStarterDesignsResponse, unknown > {
+	enabled?: boolean;
+}
+
+interface AllStarterDesignsResponse {
+	generated: { designs: StarterDesignsGenerated[] };
+	static: { designs: StarterDesignStatic[] };
+}
+
+interface StarterDesignStatic {
+	recipe: DesignRecipe;
+	slug: string;
+	title: string;
+	categories: Category[];
+	price?: string;
+}
+
+export function useStarterDesignsQuery(
+	queryParams: StarterDesignsGeneratedQueryParams,
```

Comment:
Nit: can we just duplicate the type (`StarterDesignsGeneratedQueryParams`) here in this file? As the endpoint is independent as well in the backend. Then later we can simply delete the old files. What do you think?